h/auth/cookie_authentication_policy.py

```diff
@@ -0,0 +1,60 @@
+from functools import lru_cache
+
+from h.auth.policy import IdentityBasedPolicy
+from h.security import Identity
+from h.services import AuthCookieService
```

Comment:
Yeah our hand is forced a bit by the Pyramid design. The policy is eternal and so stateless

Reply:
Ironically Pyramid has a thing called `RequestLocalCache` to let you do per-request caching without having to use `pyramid_services` ... but `RequestLocalCache` as added in Pyramid 2.0. There _are_ hacky ways to do per-request caching without `pyramid_services` in Pyramid 1. If you dig out the Pyramid PR where the idea for `RequestLocalCache` first came up one of the developers gave some examples. But we're probably better off with a service

Reply:
I just spent like 15 minutes trying to find that in the docs. You would think with `request` and `cache` this would come up.

Yeah I think this is worth a revisit post 2.0. I think we are going to have to launch a hybrid version and the move things after.